### PR TITLE
Add support for detekt

### PR DESCRIPTION
**Pull request by martinvisser**
Support detekt in pre-commit.
See https://detekt.github.io/detekt/cli.html for a list of parameters to pass.

### CODE REVIEW

File: detekt [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+BASEDIR="$(dirname "$0")"
+if [[ ! -f "$BASEDIR/detekt.jar" ]]; then
+  curl -sL -o "$BASEDIR/detekt.jar" $(curl -s https://api.github.com/repos/detekt/detekt/releases/latest | jq -r '.assets[].browser_download_url')
```
Comment (Blokje5) [inline detekt]: Probably a good idea to version pin the detekt binary, given that we had an issue with `latest` and DataDog APM recently

**Review by Blokje5 — CHANGES_REQUESTED** [review]
Mostly LGTM. If you pin the version I can approve.

File: detekt (hunk 1 above)
Comment (martinvisser) [reply detekt]: Yeah, Barry said the same. I'm going to change it to use the now latest, with an option to provide your own version

**Review by Blokje5 — APPROVED** [review]
LGTM

**Pull request merged**